src/consensus/mod.rs

```diff
@@ -0,0 +1,60 @@
+use futures::{FutureExt, Stream, StreamExt};
+
+mod paxos;
+
+use paxos::Paxos;
+
+use crate::{
+    error::{Error, Result},
+    transport::{
+        proto::{self, Endpoint},
+        Broadcast, Client, Request, Response,
+    },
+};
+
+pub struct FastPaxos<'a, C, B> {
+    broadcast: &'a mut B,
+    my_addr: Endpoint,
+    size: usize,
+    paxos: Paxos<'a, C>,
+}
+
+impl<'a, C, B> FastPaxos<'a, C, B> {
+    pub fn new(
+        my_addr: Endpoint,
+        size: usize,
+        client: &'a mut C,
+        broadcast: &'a mut B,
+    ) -> FastPaxos<'a, C, B>
+    where
+        C: Client,
+        B: Broadcast,
+    {
+        FastPaxos {
+            my_addr: my_addr.clone(),
+            broadcast,
+            size,
+            paxos: Paxos::new(client, size, my_addr),
+        }
+    }
+
+    pub async fn propose(&self, proposal: Vec<Endpoint>) {
+        // TODO: register proposal with regular paxos instance
+        unimplemented!();
+    }
+
+    pub async fn handle_message(&self, request: Request) -> Result<Response> {
```

Comment:
I feel like this relation should be the other way, `Paxos` contains `FastPaxos` and can defer to it internally?